documentation/pwa/tutorial-handle-deferred-call-result.asciidoc

```diff
@@ -10,41 +10,47 @@ When deferring an endpoint call, its result is only available after the actual s
 
 By default, Vaadin sends the deferred calls without prompting or notifying the user, and their returned result is not available in the caller view. However, often applications still need to handle deferred call results, for example, to notify the user of sending the data collected during prior offline work.
 
-This article describes how to handle deferred call results on the client side.
+This article describes how to customise deferred call handling on the client side.
 
-== The `onDeferredCall` callback
+== The `deferredCallHandler` property
 
-Since the result is not available right away for deferred calls, Vaadin has `onDeferredCall` callback property on the `ConnectClient` TypeScript endpoint client class, which allows handling deferred call results.
+Vaadin has `deferredCallHandler` property on the `ConnectClient` TypeScript endpoint client class, which allows customising the deferred call handling. 
 
-To define a custom `onDeferredCall` callback, import the client and set the property in your `index.ts` file:
+The main philosophy for customization is to wrap the deferred call submission in a try-catch block, so that you can handle it correspondingly when a deferred call submission succeeds or fails.
+
+By default, Vaadin will remove the deferred call from the queue when submitted successfully or if the submission error is handled already, for example, in a try-catch. To keep the deferred call in the queue, you can call `deferredCallSubmitter.keepDeferredCallInTheQueue()` in the catch block.
+
+To define a custom `deferredCallHandler` callback, import the client and set the property in your **`index.ts`** file as below. It is recommended to define the `deferredCallHandler` in the `index.ts` because if you put it in a sub view, the view component might not be available when the deferred call handling occurs.
 
 [source,typescript]
 .index.ts excerpt
 ----
 import client from './generated/connect-client.default';
-import {EndpointRequest} from "./Connect";
-
-client.onDeferredCall = async (call: EndpointRequest, resultPromise: Promise<any>) => {
-  try {
-    // Await the result
-    const result = await resultPromise;
-
-    // Notify the user of successful result
-  } catch (error) {
-    // Notify the user of the error
-
-    // The call will be removed from the deferred queue
-    // when sent successfully or if the error is handled here.
-
-    // Re-throw the error to keep the call in the queue:
-    // throw error;
+import { DeferredCallSubmitter } from '@vaadin/flow-frontend';
+
+client.deferredCallHandler = {
+  async handleDeferredCallSubmission(deferredCallSubmitter: DeferredCallSubmitter) {
+    try{
```

Comment:
Maybe reformat this with spaces between the keywords (`try`, `catch`) and `{`/`}`.

Reply:
done